src/internal/http/DescopeClient.swift

```diff
@@ -493,6 +494,16 @@ class DescopeClient: HTTPClient {
     }
 }
 
+func baseURLForProjectId(_ projectId: String) -> String {
+    let prefix = "https://api"
+    let suffix = "descope.com"
+    guard projectId.count >= 32 else { return "\(prefix).\(suffix)" }
+    let start = projectId.index(projectId.startIndex, offsetBy: 1)
+    let end = projectId.index(start, offsetBy: 3)
```

Comment:
In most place we protect by -27 from the end, @talaharoni so you think it can bite us later?

Reply:
why? not sure i'm following

Reply:
@orius123 This code asserts the project ID is longer or equal to 32, and if it us, it extracts the region from the designated characters. What's the problem you're talking about?

Reply:
for now this implementation will work.. i think it's better to have the same logic to remove the last 27 chars just in case we would like to change something in the future.

Reply:
Yes, also think it should be consistent with other SDKs, this way it will be easier in the future to add / remove stuff form the ID

Reply:
I'm not following, what do you mean exactly? The index is calculated from the end by removing 27 characters and not from the beginning?

Reply:
Wow, this is way longer that we should have discussed it :)
Let's go this way and I will adapt the node and go sdks

Reply:
Next time come to the office 😀